README.md

```diff
@@ -21,7 +26,66 @@ containing ascii characters):
 assert(scopeUtils.validScope("..."));
 ```
 
-### Checking Scope Sets
+### Scope Expressions
+
+These are the "new-style" way of dealing with scopes and allow for greater
+flexibility than the old style. The expressions are objects with a single key --
+either `AnyOf` or `AllOf`. The value must be an array and the elements of the
+array must be either expressions or a [valid scope](#valid-scopes).
+
+A scope expression can be evaluated against an array of scopes to determine if the
+scope expression is "satisfied" by the array of scopes. Satisfaction in this context
+means that enough scopes in the expression are either in the array directly or are
+prefixed by a scope in the array that ends in `*` to allow the `AnyOf` and `AllOf`
+clauses to be true.
```

Comment:
I think this should refer to "normal" scope satisfaction as documented in https://docs.taskcluster.net/reference/platform/taskcluster-auth/docs/scopes -- that will avoid needing to mention `*`.

Also, you can use "scope set" instead of "array of scopes".  The weird use of terminology was the singular "scopesets" to refer to nested arrays, but that's now deprecated.

Reply:
The `validExpression` function isn't mentioned anywhere, and `satisfiesExpression` only appears in the examples!